test/python/pulse/test_commands.py

```diff
@@ -244,5 +245,162 @@ def test_can_construct_discriminator_with_default_values(self):
+        pulse_lib_gaus = pulse_lib.gaussian(duration=25, sigma=4,
+                                            amp=0.5j, zero_ends=False).samples
+        np.testing.assert_almost_equal(sample_pulse.samples, pulse_lib_gaus)
+
+    def test_gauss_samples(self):
+        """Test that the gaussian samples match the formula."""
+        duration = 25
+        sigma = 4
+        amp = 0.5j
+        # formulaic
+        times = np.array(range(25), dtype=np.complex_)
+        times = times - (duration / 2) + 0.5
+        gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        # command
+        command = Gaussian(duration=duration, sigma=sigma, amp=amp)
+        samples = command.get_sample_pulse().samples
+        np.testing.assert_almost_equal(samples, gauss)
+
+    def test_gauss_square_samples(self):
+        """Test that the gaussian square samples match the formula."""
+        duration = 125
+        sigma = 4
+        amp = 0.5j
+        # formulaic
+        times = np.array(range(25), dtype=np.complex_)
+        times = times - (25 / 2) + 0.5
+        gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        # command
+        command = GaussianSquare(duration=duration, sigma=sigma, amp=amp, width=100)
+        samples = command.get_sample_pulse().samples
+        np.testing.assert_almost_equal(samples[50], amp)
+        np.testing.assert_almost_equal(samples[100], amp)
+        np.testing.assert_almost_equal(samples[:10], gauss[:10])
+
+    def test_gauss_square_extremes(self):
+        """Test that the gaussian square pulse can build a gaussian."""
+        duration = 125
+        sigma = 4
+        amp = 0.5j
+        gaus_square = GaussianSquare(duration=duration, sigma=sigma, amp=amp, width=0)
+        gaus = Gaussian(duration=duration, sigma=sigma, amp=amp)
+        np.testing.assert_almost_equal(gaus_square.get_sample_pulse().samples,
+                                       gaus.get_sample_pulse().samples)
+        gaus_square = GaussianSquare(duration=duration, sigma=sigma, amp=amp, width=121)
+        const = ConstantPulse(duration=duration, amp=amp)
+        np.testing.assert_almost_equal(gaus_square.get_sample_pulse().samples[2:-2],
+                                       const.get_sample_pulse().samples[2:-2])
+
+    def test_drag_samples(self):
+        """Test that the drag samples match the formula."""
+        duration = 25
+        sigma = 4
+        amp = 0.5j
+        beta = 1
+        # formulaic
+        times = np.array(range(25), dtype=np.complex_)
+        times = times - (25 / 2) + 0.5
+        gauss = amp * np.exp(-(times / sigma)**2 / 2)
+        gauss_deriv = -(times / sigma**2) * gauss
+        drag = gauss + 1j * beta * gauss_deriv
+        # command
+        command = Drag(duration=duration, sigma=sigma, amp=amp, beta=beta)
+        samples = command.get_sample_pulse().samples
+        np.testing.assert_almost_equal(samples, drag)
+
+    def test_drag_validation(self):
```

Comment:
Can you also add one case where the validation is only broken by the additional extrema (ie the sqrt terms)?

Reply:
I have that test already (line 358) but I can add another one anyway